apps/tailwind-components/components/table/TableEMX2.vue

```diff
@@ -1,23 +1,91 @@
 <script setup lang="ts">
-import type { IColumn } from "../../meta-data-utils/src/types";
+import type { ITableSettings } from "~/types/types";
+import type { IColumn } from "../../../metadata-utils/src/types";
 
-defineProps<{
-  columns: IColumn[];
-  rows: Record<string, any>[];
-}>();
+const props = withDefaults(
+  defineProps<{
+    columns: IColumn[];
+    rows: Record<string, any>[];
+    count: number;
+    settings?: ITableSettings;
+  }>(),
+  {
+    settings: {
+      page: 1,
+      pageSize: 10,
+      orderby: { column: "", direction: "ASC" },
+      search: "",
+    },
+  }
+);
+
+const emit = defineEmits(["update:settings"]);
+
+function handleSortRequest(columnId: string) {
+  let direction = "ASC";
+  if (props.settings?.orderby?.column === columnId) {
+    direction = props.settings.orderby.direction === "ASC" ? "DESC" : "ASC";
+  }
+  emit("update:settings", {
+    ...props.settings,
+    orderby: { column: columnId, direction },
+  });
+}
+
+function handleSearchRequest(search: string) {
+  emit("update:settings", {
+    ...props.settings,
+    search,
+  });
+}
+
+function handlePagingRequest(page: number) {
+  emit("update:settings", {
+    ...props.settings,
+    page,
+  });
+}
 </script>
 <template>
-  <div class="overflow-x-auto overscroll-contain">
-    <table class="text-left bg-white table-fixed w-full">
+  <div class="flex">
+    <FilterSearch
+      class="w-2/5"
+      :modelValue="settings.search"
+      @update:modelValue="handleSearchRequest"
+      :inverted="true"
+    >
+    </FilterSearch>
+  </div>
+
+  <div class="overflow-x-auto overscroll-x-contain">
+    <table class="text-left table-fixed w-full">
       <thead>
         <tr class="">
           <th
             v-for="column in columns"
             class="py-2.5 px-2.5 border-b border-gray-200 first:pl-0 last:pr-0 sm:first:pl-2.5 sm:last:pr-2.5 text-left w-64"
           >
-            <div class="overflow-ellipsis whitespace-nowrap overflow-hidden">
+            <div
+              class="overflow-ellipsis whitespace-nowrap overflow-hidden hover:cursor-pointer"
+              @click="handleSortRequest(column.id)"
+            >
               {{ column.label }}
```

Comment:
I wonder if we should create a dropdown menu for each column that provides a number of options (e.g., sort ascending/descending, filtering, show/hide column, etc.). This will address issues with semantic HTML and ensuring the actions/labels are consistent.